src/main/java/com/google/cloud/discotoproto3converter/proto3/DocumentToProtoConverter.java

```diff
@@ -152,6 +157,50 @@ private void cleanupEnumNamingConflicts() {
     }
   }
 
+  private void convertEnumFieldsToStrings() {
+    if (!this.enumsAsStrings) {
+      return;
+    }
+
+    Message stringType = Message.PRIMITIVES.get("string");
+    for (Message message : allMessages.values()) {
+      SortedSet<Field> enumFields = new TreeSet<>();
+      for (Field field : message.getFields()) {
+        // Enums declared in Operation must remain intact
+        if (field
+            .getOptions()
+            .stream()
+            .anyMatch(a -> "google.cloud.operation_field".equals(a.getName()))) {
```

Comment:
IIUC, you're checking whether the field is inside Operation. Is it possible/simpler to simply check whether `Message` is an operation? (honest question, I don't know)

Reply:
This does not check if the field is inside an operation, it checks if there is any field in the current message has the `operation_field` annotation in it (and this is what makes that message "Operation", while it does not have to be even called "Operation"). This logic let's us separate logic which determines what is operation what is not and the enums decisions. If this logic was checkign for operation independently, then the "operation" detection logic would be duplicated: one in the logic which assigns `operation_field`annotation, and one here.